src/jsonrpc/api.rs

```diff
@@ -87,6 +87,58 @@ fn broadcast_spend(control: &DaemonControl, params: Params) -> Result<serde_json
     Ok(serde_json::json!({}))
 }
 
+fn list_coins(control: &DaemonControl, params: Option<Params>) -> Result<serde_json::Value, Error> {
+    let statuses_arg = params
+        .as_ref()
+        .and_then(|p| p.get(0, "statuses"))
+        .and_then(|statuses| statuses.as_array());
+    let mut statuses: Vec<CoinStatus> = vec![];
+    if let Some(statuses_arg) = statuses_arg {
+        for status_arg in statuses_arg.iter() {
+            statuses.push(
+                status_arg
+                    .as_str()
+                    .and_then(CoinStatus::from_arg)
+                    .ok_or_else(|| {
+                        Error::invalid_params(format!(
+                            "Invalid value {} in 'statuses' parameter.",
+                            status_arg
+                        ))
+                    })?,
+            );
+        }
+    }
+    let outpoints_arg = params
+        .as_ref()
+        .and_then(|p| p.get(1, "outpoints"))
+        .and_then(|op| op.as_array());
+    let mut outpoints: Vec<bitcoin::OutPoint> = vec![];
+    if let Some(outpoints_arg) = outpoints_arg {
+        for op_arg in outpoints_arg.iter() {
+            outpoints.push(
+                op_arg
+                    .as_str()
+                    .ok_or_else(|| {
+                        Error::invalid_params(format!(
+                            "Invalid value {} in 'outpoints' parameter.",
+                            op_arg
+                        ))
+                    })
+                    .map(|op| {
+                        bitcoin::OutPoint::from_str(op).map_err(|_| {
+                            Error::invalid_params(format!(
+                                "Invalid value {} in 'outpoints' parameter.",
+                                op_arg
+                            ))
+                        })
+                    })??,
+            );
+        }
+    }
```

Comment:
There might be a better way of writing this, but this was the best I could get working :blush:

Reply:
I think I've found a better way :nerd_face: Will test it and then commit shortly.

Reply:
I've now updated this.